app/assets/javascripts/admin/products/services/variant_unit_manager.js.coffee

```diff
@@ -30,7 +48,16 @@ angular.module("admin.products").factory "VariantUnitManager", ->
         unitScales[0]
 
     @getUnitName: (scale, unitType) ->
-      @unitNames[unitType][scale]
+      if @units[unitType][scale]
+        @units[unitType][scale]['name']
+      else
+        ''
 
     @unitScales: (unitType) ->
-      (parseFloat(scale) for scale in Object.keys(@unitNames[unitType])).sort()
+      (parseFloat(scale) for scale in Object.keys(@units[unitType])).sort (a, b) ->
```

Comment:
Nice. I had to look this up, and apparently javascript `sort()` doesn't play nicely with numbers by default unless the compare function is given explicitly, as you've done here. :+1:

Reply:
Haha yeah I had to do the same. Silly javascript.